java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -49,6 +50,15 @@
  * Factory for creating ORC tree readers.
  */
 public class TreeReaderFactory {
+  // The current JDK has a bug where values of the form:
+  // YYYY-MM-DD HH:MM:SS.000X is off by a second for dates before 1970.
```

Comment:
Can you please link the BUG # for future reference (can be done on commit)? 
Also can we run the tests against JDK version with the bug on travis CI to trigger TIMSTAMP_BUG == true code path? The logic for adjusting millis look good to me.